File: ePrijevozSarajevo/ePrijevozSarajevo.Model/Ticket.cs
namespace ePrijevozSarajevo.Model
{
    public class Ticket
    {
        public int TicketId { get; set; }
        public string Name { get; set; } = null!;
        public double Price { get; set; }
        public string? StateMachine { get; set; }
        public DateTime? ModifiedDate { get; set; }
        public int? CurrentUserId { get; set; }
    }
}

File: ePrijevozSarajevo/UI/eprijevoz_desktop/lib/models/ticket.dart
import 'package:json_annotation/json_annotation.dart';

part 'ticket.g.dart';

@JsonSerializable()
class Ticket {
  int? ticketId;
  String? name;
  double? price;
  String? stateMachine;
  List<String>? allowedActions = [];
  DateTime? modifiedDate;

  Ticket({this.ticketId, this.name, this.price, this.stateMachine});

  factory Ticket.fromJson(Map<String, dynamic> json) => _$TicketFromJson(json);

  Map<String, dynamic> toJson() => _$TicketToJson(this);
}
